### DISCUSSION TITLE
[AI-Deck] Is there a crazyflie-firmware source (can be compiled and flashed by myself) that is supposed to work properly with the ai-deck?

### DISCUSSION THREAD

**Discussion by chengliu-LR** [bitcraze/discussions#181] [Q&A] [answered]
Hi all,

Currently I am working with ai-deck and have finished the tutorials and prerequisites on this website [WiFi Video Streamer](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/test-functions/wifi-streamer/). The wifi demo works well with the crazyflie-firmware that I updated using `cfclient`. However, when I try to use the firmware built from [crazyflie-firmware source](https://github.com/bitcraze/crazyflie-firmware) (master branch), the wifi-streaming demo is not working anymore. The reason seems like the crazyflie-firmware can't detect the GAP8 and ESP32 with the built-from-source version. And when I try to flash the wifi-streaming demo again to the ai-deck through USB dongle based on this built-from-source crazyflie-firmware, the flashing will be stuck. So I suppose, the crazyflie-firmware updated with `cfclient` is not the same version as the latest built-from-source crazyflie-firmware.

Since I need to use both of the `app layers` in the crazyflie-firmware and the ai-deck at the same time, I really need to build the crazyflie-firmware and flash it to crazyflie by myself. **So my question is**: Is there a crazyflie-firmware source version that is supposed to work properly with the ai-deck?

Thank you!

**Comment by knmcguire** [comment, marked as answer]
Hi!

If you have updated the CF firmware to the latest development branch, you need to update the ESP as well due to the changed baudrate of the uart connection. Somebody else also encountered this a while ago, so this should have the answer to your questions:

https://github.com/orgs/bitcraze/discussions/151
